src/write_influxdb_udp.c

```diff
@@ -365,136 +365,131 @@ static int wifxudp_escape_string(char *buffer, size_t buffer_size,
                                    __VA_ARGS__);                               \
-    if (status < 0)                                                            \
-      return -1;                                                               \
+    if (status < 0) {                                                          \
+      return metric_offset;                                                    \
+    }                                                                          \
     offset += status;                                                          \
   } while (0)
 
 #define BUFFER_ADD(...)                                                        \
   do {                                                                         \
     status = snprintf(buffer + offset, buffer_len - offset, __VA_ARGS__);      \
     if ((status < 0) || (status >= (buffer_len - offset))) {                   \
-      sfree(rates);                                                            \
-      return -1;                                                               \
+      return metric_offset;                                                    \
     }                                                                          \
     offset += status;                                                          \
   } while (0)
 
-  BUFFER_ADD_ESCAPE(vl->plugin);
-  BUFFER_ADD(",host=");
-  BUFFER_ADD_ESCAPE(vl->host);
-  if (strcmp(vl->plugin_instance, "") != 0) {
-    BUFFER_ADD(",instance=");
-    BUFFER_ADD_ESCAPE(vl->plugin_instance);
-  }
-  if (strcmp(vl->type, "") != 0) {
-    BUFFER_ADD(",type=");
-    BUFFER_ADD_ESCAPE(vl->type);
-  }
-  if (strcmp(vl->type_instance, "") != 0) {
-    BUFFER_ADD(",type_instance=");
-    BUFFER_ADD_ESCAPE(vl->type_instance);
-  }
+  for (; *idx < fam->metric.num; (*idx)++) {
+    int offset = metric_offset;
+    have_values = false;
+    BUFFER_ADD_ESCAPE(fam->name);
+    metric_t metric = fam->metric.ptr[*idx];
 
-  BUFFER_ADD(" ");
-  for (size_t i = 0; i < ds->ds_num; i++) {
-    if ((ds->ds[i].type != DS_TYPE_COUNTER) &&
-        (ds->ds[i].type != DS_TYPE_GAUGE) &&
-        (ds->ds[i].type != DS_TYPE_DERIVE)) {
-      sfree(rates);
-      return -1;
+    for (size_t j = 0; j < metric.label.num; j++) {
+      label_pair_t label = metric.label.ptr[j];
+      if (strcmp(label.name, "") != 0 && strcmp(label.value, "") != 0) {
+        BUFFER_ADD(",");
+        BUFFER_ADD_ESCAPE(label.name);
+        BUFFER_ADD("=");
+        BUFFER_ADD_ESCAPE(label.value);
+      }
     }
+    BUFFER_ADD(" ");
 
-    if (ds->ds[i].type == DS_TYPE_GAUGE) {
-      if (isnan(vl->values[i].gauge))
-        continue;
-      if (have_values)
-        BUFFER_ADD(",");
-      BUFFER_ADD("%s=%lf", ds->ds[i].name, vl->values[i].gauge);
-      have_values = true;
+    if (fam->type == METRIC_TYPE_GAUGE) {
+      if (!isnan(metric.value.gauge)) {
+        BUFFER_ADD("value=%lf", metric.value.gauge);
```

Comment:
Sorry for picking on code you haven't introduced: `%lf` is uncommon, because `gauge_t` is essentially a double. I'd prefer to use `GAUGE_FORMAT` here, or `%f` if the format produced by `GAUGE_FORMAT` can't be parsed by InfluxDB.

Reply:
Ok, changed to `GAUGE_FORMAT`